Updated the building connectors documentation (#80)
ConduitIO/conduit-site: docs/connectors/building-connectors/using-a-custom-connector.mdx

---
title: "Using a Custom Connector"
sidebar_position: 5
---

## Adding the Connector to Conduit

> The Conduit Connector template includes a `Makefile` that can be used to generate a binary version of your connector

Reviewer: ```suggestion
The Conduit Connector template includes a `Makefile` that can be used to build a binary version of your connector.
```

```bash
make build
```

After the command runs you should see a binary file with the name of the connector in the local directory.

> To run the Connector, you will need to put it in a directory that Conduit is able to access.  By default, this directory is the `connectors` directory located alongside the Conduit binary.  Move your newly created Connector binary into the `connectors` directory and run Conduit using the following command:

Reviewer: ```suggestion
To run the Connector, you will need to move it to a directory that Conduit can access.  By default, this directory is the `connectors` directory located alongside the Conduit binary.  Move your newly created Connector binary into the `connectors` directory and run Conduit using the following command:
```

```bash
./conduit
```

Navigate to the Conduit UI (by default located at [http://localhost:8080/ui](http://localhost:8080/ui) while Conduit is running) to see your Connector in the list of available Connectors

> ![Custom Connector in Conduit UI](/img/conduit-custom-connector-ui-screenshot.png)

Reviewer: noticed we still have the other images with code snippets and they're not used anymore 👀

Reviewer: @William-Hill No you got them all! you just forgot to delete the other screenshots from the images folder :D

## Using the Connector in a Conduit Pipeline

Although you can use the Connector directly within the Conduit UI, the recommended way to use the Connector is define a [pipeline configuration file](/docs/pipeline-configuration-files/getting-started.mdx).

> When using custom Connectors in a pipeline configuration, they are prefixed with phrase 'standalone' unlike built in Connectors which are prefixed with 'builtin'. For example, to use a custom Connector named `file-sync` it would look as follows: `plugin: standalone:file-sync@[replace-with-connector-version]`

Reviewer: you could also link to this page https://conduit.io/docs/connectors/referencing for more info about referencing connectors

Author: Made suggested change here: https://github.com/ConduitIO/conduit-site/pull/80/commits/0b2f7c2d1a55a6c43982ba713262a7ef31a76e5b
